### PR TITLE
Use scm setuptools for auto-versionning

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Description

Move to scm_setuptools for easier release. Update the release doc.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :sparkles: New feature |
| ✓   | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Related Issue

#4399 #4400

### CODE REVIEW

File: setup.cfg
Comment (cdce8p) [inline setup.cfg:50]: According to the doc, moving this here to the `pyproject.toml` file would be the preferred. https://github.com/pypa/setuptools_scm#pyprojecttoml-usage

**Review by cdce8p — COMMENTED** [review]
Tbh I'm not too familiar with `setuptools_scm`. If you think it will make things easier, go for it.

File: setup.cfg
Comment (Pierre-Sassoulas) [reply setup.cfg:50]: Yeah, but we don't have ``pyproject.toml`` yet :D

**Comment by Pierre-Sassoulas** [comment]
Basically it removes the need to change the version and is based on tags only. So no more change to do in the version value before and after release.

File: setup.cfg
Comment (cdce8p) [reply setup.cfg:50]: True 🤷🏻‍♂️

**Comment by wj-Mcat** [comment]
Wating for this pr to be merged so my github ci can pass 👻

**Comment by Pierre-Sassoulas** [comment]
I don't understand @wj-Mcat ? Why would it fix your CI ?

**Comment by Pierre-Sassoulas** [comment]
@cdce8p I think we can do that in 2.8 directly and we don't have to change the tags either, scm_setuptools seems to understand the tag.

**Comment by cdce8p** [comment]
> @cdce8p I think we can do that in 2.8 directly and we don't have to change the tags either, scm_setuptools seems to understand the tag.

Not exactly sure what you mean here. In my experience the tag is usually just something like `v1.x.x`, but if the current naming schema works there is no reason to change it.

**Comment by Pierre-Sassoulas** [comment]
> Not exactly sure what you mean here.

I wanted to change the tag from ``pylint-1.2.3`` to ``v1.2.3`` but the generated version by setuptool_scm is ``1.2.3`` for both. So we can keep the standard. We might still want to change the tag to something standard for pre-commit the version will still be ``pylint-1.2.3`` in .pre-commitconf.yml...

**Comment by cdce8p** [comment]
> > Not exactly sure what you mean here.
> 
> I wanted to change the tag from `pylint-1.2.3` to `v1.2.3` but the generated version by setuptool_scm is `1.2.3` for both. So we can keep the standard. We might still want to change the tag to something standard for pre-commit the version will still be `pylint-1.2.3` in .pre-commitconf.yml...

Ah, I see. I would recommend to use `v1.2.3` then.

File: doc/release.md
Comment (cdce8p) [inline doc/release.md:26]: Only if you what to change the tag schema

**Review by cdce8p — APPROVED** [review]

**Comment by wj-Mcat** [comment]
> I don't understand @wj-Mcat ? Why would it fix your CI ?

I use pylint to test my project, but it occured [error](https://github.com/wechaty/python-wechaty-puppet-service/runs/2430773196#step:5:35) when install new version of `pylint`.

So, I'm waiting for this pr to be merged.

**Comment by Pierre-Sassoulas** [comment]
Ho okay, it's actually another problem fixed in #4407 but I'm going to release 2.8.2 just after merging this one (that just speed thing up for us maintainers) so you're still right :)

**Pull request merged**
